src/Catalog/CatalogIndexEntry.cs

```diff
@@ -11,134 +11,111 @@
             }
+
+            Id = id;
+            Version = version ?? throw new ArgumentNullException(nameof(version));
         }
 
-        [JsonProperty("nuget:id")]
-        public string Id { get; private set; }
+        [JsonProperty("@id")]
+        [JsonRequired]
+        public Uri Uri { get; private set; }
 
-        [JsonProperty("nuget:version")]
-        public NuGetVersion Version { get; private set; }
+        [JsonProperty("@type")]
+        [JsonRequired]
+        [JsonConverter(typeof(CatalogTypeConverter))]
+        public IEnumerable<string> Types { get; private set; }
 
         [JsonProperty("commitId")]
+        [JsonRequired]
         public string CommitId { get; private set; }
 
         [JsonProperty("commitTimeStamp")]
+        [JsonRequired]
         public DateTime CommitTimeStamp { get; private set; }
 
-        public int CompareTo(CatalogIndexEntry other)
-        {
-            return CommitTSComparer.Compare(this, other);
-        }
+        [JsonProperty("nuget:id")]
+        [JsonRequired]
+        public string Id { get; private set; }
 
-        public bool IsDelete()
-        {
-            return _type == "nuget:PackageDelete";
-        }
+        [JsonProperty("nuget:version")]
+        [JsonRequired]
+        public NuGetVersion Version { get; private set; }
 
-        // common comparers for sorting and creating sets from these entries
-        public static CatalogIndexEntryIdComparer IdComparer
-        {
-            get
-            {
-                return new CatalogIndexEntryIdComparer();
-            }
-        }
+        [JsonIgnore]
+        public bool IsDelete { get; }
 
-        public static CatalogIndexEntryIdVersionComparer IdVersionComparer
+        public int CompareTo(CatalogIndexEntry other)
         {
-            get
+            if (other == null)
             {
-                return new CatalogIndexEntryIdVersionComparer();
+                throw new ArgumentNullException(nameof(other));
             }
+
+            return _commitTimeStampComparer.Compare(this, other);
         }
 
-        public static CatalogIndexEntryDateComparer CommitTSComparer
+        public static CatalogIndexEntry Create(JToken token)
         {
-            get
+            if (token == null)
             {
-                return new CatalogIndexEntryDateComparer();
+                throw new ArgumentNullException(nameof(token));
             }
-        }
-    }
-
-    public class CatalogIndexEntryDateComparer : IComparer<CatalogIndexEntry>
-    {
-        public int Compare(CatalogIndexEntry x, CatalogIndexEntry y)
-        {
-            return x.CommitTimeStamp.CompareTo(y.CommitTimeStamp);
-        }
-    }
 
+            var uri = new Uri(token["@id"].ToString());
+            var type = token["@type"].ToString();
+            var commitId = token["commitId"].ToString();
+            var commitTimeStamp = DateTime.ParseExact(
+                token["commitTimeStamp"].ToString(),
```

Comment:
The reason for this change is that tests failed on my dev machine because my machine's current time zone is _not_ UTC.  All our build and job machines run in UTC, so this change is about ensuring consistency wherever the code is executed.  We've had similar time zone bugs before ([example 1](https://github.com/NuGet/NuGetGallery/issues/6187) and [example 2](https://github.com/NuGet/NuGetGallery/issues/6189#issuecomment-406408036)).

This issue wasn't caught until now because there simply weren't tests for `CatalogIndexEntry`.

It's one thing that API documentation doesn't specify the exact datetime format.  It's another thing to ensure that the implementation does not change in ways that are not well understood.  Feed2Catalog tests ([here](https://github.com/NuGet/NuGet.Services.Metadata/blob/c9f960202abeffe6be1977686d4024f31e42be91/tests/NgTests/Feed2CatalogTests.cs#L1106) and [here](https://github.com/NuGet/NuGet.Services.Metadata/blob/c9f960202abeffe6be1977686d4024f31e42be91/tests/NgTests/Feed2CatalogTests.cs#L1245)) verify that commit timestamps follow a specific format, and I expect those tests to fail if the format changes.